extensions/resource-deployment/src/ui/resourceTypeWizard.ts

```diff
@@ -0,0 +1,157 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { DeploymentProvider, instanceOfWizardDeploymentProvider, ResourceType } from '../interfaces';
+import { Model } from './model';
+import { WizardPageBase } from './wizardPageBase';
+import * as nls from 'vscode-nls';
+import { DeployClusterWizardModel } from './deployClusterWizard/deployClusterWizardModel';
+import { WizardPageInfo } from './wizardPageInfo';
+import { IKubeService } from '../services/kubeService';
+import { IAzdataService } from '../services/azdataService';
+import { INotebookService } from '../services/notebookService';
+import { IToolsService } from '../services/toolsService';
+import { IPlatformService } from '../services/platformService';
+const localize = nls.loadMessageBundle();
+
+
+export class ResourceTypeWizard {
+	private customButtons: azdata.window.Button[] = [];
+	public pages: ResourceTypePage[] = [];
+	public wizardObject: azdata.window.Wizard;
+	public toDispose: vscode.Disposable[] = [];
+	public model: ResourceTypeModel;
+	private _useGenerateScriptButton!: boolean;
+
+	public get useGenerateScriptButton(): boolean {
+		return this._useGenerateScriptButton;
+	}
+
+	public set useGenerateScriptButton(value: boolean) {
+		this._useGenerateScriptButton = value;
+	}
+
+	//TODO: eventually only resourceType will be passed. For now, we are passing both the resourceType and provider
+	constructor(
+		public resourceType: ResourceType,
+		public provider: DeploymentProvider,
+		public _kubeService: IKubeService,
+		public azdataService: IAzdataService,
+		public notebookService: INotebookService,
+		public toolsService: IToolsService,
+		public platformService: IPlatformService) {
+		this.wizardObject = azdata.window.createWizard(resourceType.displayName, resourceType.name, 'wide');
+		this.model = this.getResourceProviderModel();
+	}
+
+
+	public getResourceProviderModel(): ResourceTypeModel {
```

Comment:
If this is returning undefined then the return type should be | undefined

Reply:
For now, yes. Later on when all providers are migrated, it won't return undefined

Reply:
You should still make sure the return types match what can actually be returned - even if it's temporary. You can remove it later on if needed.